Microsoft.Toolkit/Collections/ObservableGroupedCollectionExtensions.cs

```diff
@@ -22,8 +26,18 @@ public static class ObservableGroupedCollectionExtensions
         /// <param name="key">The key of the group to query.</param>
         /// <returns>The first group matching <paramref name="key"/>.</returns>
         /// <exception cref="InvalidOperationException">The target group does not exist.</exception>
+        [Pure]
         public static ObservableGroup<TKey, TValue> First<TKey, TValue>(this ObservableGroupedCollection<TKey, TValue> source, TKey key)
-            => source.First(group => GroupKeyPredicate(group, key));
+        {
+            ObservableGroup<TKey, TValue>? group = source.FirstOrDefault(key);
+
+            if (group is null)
+            {
+                ThrowArgumentExceptionForKeyNotFound();
```

Comment:
@Sergio0694 Guard API or no?

Reply:
I thought it might be best for us not to use that in this case so that we could have a clearer error message for users. As in, this way we'll get "The desired key is not present in the collection" rather than a more obscure "Parameter group must not be null" (like, what group? Where?).